src/wallet/wallet.cpp

```diff
@@ -2546,6 +2548,16 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
     // all.
     txNew.nLockTime = std::max(0, chainActive.Height() - 10);
 
+    // Set nExpiryHeight to 20 blocks (default) past current blockheight
+    // nVersion of Overwinter transactions that expire need to be version 3
+    txNew.nVersion = 3;
```

Comment:
We can't do this before Overwinter activates, so gate this entire block in a `NetworkUpgradeActive` check. Also, I expect the `txNew.nVersion = 3` here will become redundant, as #2906 itself will need to update the wallet code to set this.